src/initialization/InitDistribution.cpp

```diff
@@ -121,6 +125,91 @@ namespace impactx
         }
     }
 
+    void initialization::setDistributionParametersFromTwissInputs (
+        amrex::ParmParse const & pp_dist,
+        amrex::ParticleReal& sigx, amrex::ParticleReal& sigy, amrex::ParticleReal& sigt,
+        amrex::ParticleReal& sigpx, amrex::ParticleReal& sigpy, amrex::ParticleReal& sigpt,
+        amrex::ParticleReal& muxpx, amrex::ParticleReal& muypy, amrex::ParticleReal& mutpt
+    )
+    {
+
+        // Values to be read from input
+        amrex::ParticleReal betax, betay, betat, emittx, emitty, emittt;
+        // If alpha is zero the bunch is in focus
+        amrex::ParticleReal alphax = 0.0, alphay = 0.0, alphat = 0.0;
+
+        // Reading the input Twiss parameters
+        pp_dist.query("alphaX", alphax);
+        pp_dist.query("alphaY", alphay);
+        pp_dist.query("alphaT", alphat);
+        pp_dist.get("betaX", betax);
+        pp_dist.get("betaY", betay);
+        pp_dist.get("betaT", betat);
+        pp_dist.get("emittX", emittx);
+        pp_dist.get("emittY", emitty);
+        pp_dist.get("emittT", emittt);
+
+        if (betax <= 0.0_prt || betay <= 0.0_prt || betat <= 0.0_prt) {
+            amrex::Abort("Input Error: The beta function values need to be non-zero positive values in all dimensions.");
```

Comment:
Please throw a runtime exception instead. This is input, so a user might want to recover in a Python notebook from it.

AMReX' abort does something more complex: https://amrex-codes.github.io/amrex/docs_html/Basics.html?highlight=amrex+abort#abort-assertion-and-backtrace